src/classLibrary/Generator.ts

```diff
@@ -0,0 +1,121 @@
+import { CyanSafe, 
+    Glob, 
+    IFileSystemInstanceMetadata, 
+    IGuidGenerator, 
+    IParsingStrategy, 
+    VirtualFileSystemInstance } from "./interfaces/interfaces";
+import { Parser } from "./Parser";
+import { GuidResolver } from "./ParsingStrategies/GuidResolver";
+import { IfElseResolver } from "./ParsingStrategies/IfElseResolver";
+import { InlineFlagResolver } from "./ParsingStrategies/InlineFlagResolver";
+import { VariableResolver } from "./ParsingStrategies/VariableResolver";
+import { Utility } from "./Utility/Utility";
+import fs from "fs";
+import path from "path";
+import { GlobFactory } from "./Utility/GlobFactory";
+import chalk from "chalk";
+import { FileFactory } from "./Utility/FileFactory";
+
+import deleteEmpty from "delete-empty";
+import { PackageResolver } from "./ParsingStrategies/PackageResolver";
+import { PluginHandler } from "./PluginHandler";
+
+export class Generator {
+    private readonly util: Utility;
+    private guidGenerator: IGuidGenerator;
+
+    constructor(util: Utility, guidGenerator: IGuidGenerator) {
+        this.util = util;
+        this.guidGenerator = guidGenerator;
+    }
+    
+    async GenerateVFS(cyanSafe: CyanSafe, globFactory: GlobFactory): Promise<VirtualFileSystemInstance[]> {
+    
+        //package resolver may need to think how to expand it to beyond npm modules
+        let strategies: IParsingStrategy[] = [
+            new GuidResolver(this.guidGenerator, this.util),
+            new IfElseResolver(this.util),
+            new InlineFlagResolver(this.util),
+            new VariableResolver(this.util),
+            new PackageResolver(this.util)
+        ];
+
+        let parser: Parser = new Parser(this.util, strategies, cyanSafe);
+
+        console.log(chalk.greenBright("Preparation done!"));
+        console.log(chalk.cyanBright("Performing variable and flag scans..."));
+
+        //create globs (empty content vfs)
+        let filesMetadata: IFileSystemInstanceMetadata[] = cyanSafe.globs.Map((g: Glob) => globFactory.GenerateFilesMetadata(g, g.root)).Flatten();
```

Comment:
I think instead of returning a `IFileSystemInstanceMetadata` this should return a `VirtualFileSystemInstance` instead, with reference to the method `CreateFileSystemInstance` under `RootFileFactory.ts` in old cyanprint.